botorch/acquisition/analytic.py

```diff
@@ -470,6 +470,14 @@ def _preprocess_constraint_bounds(
         self.register_buffer("con_lower", torch.tensor(con_lower, dtype=torch.float))
         self.register_buffer("con_upper", torch.tensor(con_upper, dtype=torch.float))
 
+        # All this is to make Pyre happy
```

Comment:
:(

Reply:
Ideally this wouldn't be necessary, not sure though whether that's possible

Reply:
New version, hopefully less ugly

Reply:
New version. Still verbose, but more explicit about what's going on and less redundant.

Reply:
> I think it should be possible to do all of this in one line per assignment with self.con_lower = torch.tensor(con_lower) and it should be registered as a buffer.

This only works for modules, not for buffers. To make something a buffer, you actually have to use `register_buffer` (not sure if there is a way to make a Buffer object or sth like that and then assign this though). The reason for this is that there are plenty of cases where you'll want to make a tensor an instance attribute but not a buffer (so that it doesn't get added to the `state_dict`). The reason we have these tensors as buffer here is so that a `module.to(dtype=, device=)` call automatically moves them over to the right device/dtype if needed.